src/logfile_values_experimental.py

```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/env python
+#
+# logfile_values.py
+#
+# Copyright (c) 2017, InnoGames GmbH
+#
+"""
+logfile_values.py -- a python script to find metrics values in log file
+
+This script can retrive last line values by column number:
+
+python logfile_values.py --metric "metric1:1" "metric2:2" ...
+
+And aggregate data by time period with different functions:
+
+python logfile_values.py  --metric "metric1:1:mean:1d" --metric "metric2:3:count:60s"
+
+"""
+import re
+import time
+import os
+import gzip
+import logging
+
+from argparse import ArgumentParser, ArgumentTypeError
+from datetime import timedelta
+
+
+class Metric:
+    def __init__(self, arg):
+        if ':' not in arg:
+            raise ArgumentTypeError('Argument must have ":"')
+        parts = arg.split(':')
+        if len(parts) != 4 and len(parts) != 2:
+            raise ValueError('Wrong number of options')
+
+        # Accepts 2 or 4 options. If 2 then function and period are NULL
+        parts += [None] * (4 - len(parts))
+        self.name, column, function, period = parts
+        if period:
+            pattern = re.compile("^\d+[A-Za-z]{1,3}$")
+            if not pattern.match(period):
+                raise ArgumentTypeError('Period must have number and unit')
+        if not column.isdecimal():
+            raise ArgumentTypeError('Column must be a number')
+        self.column = int(column)
+        self.function = function
+        self.period = period
+        # Here is container for metric values
+        self.values = []
+        self.last_value = 0
+        self.now = int(time.time())
+
+    def get_timeshift(self):
+        if self.period:
+            value = int(self.period[:-1])
+            unit = self.period[-1].lower()
```

Comment:
This wouldn't work now as the unit is not always 1 character.  I am sorry for causing more work to you.  Standards are something we never sacrifice.